compiler/infra/BitVector.hpp

```diff
@@ -156,25 +156,68 @@ class TR_BitVector
 
    // Construct an empty bit vector. All bits are initially off.
    //
-   TR_BitVector() : _numChunks(0), _chunks(NULL), _firstChunkWithNonZero(0), _lastChunkWithNonZero(-1), _allocationKind(heapAlloc), _growable(growable), _trMemory(0) { }
+   TR_BitVector() : _numChunks(0), _chunks(NULL), _firstChunkWithNonZero(0), _lastChunkWithNonZero(-1), _growable(growable), _region(0) { }
+   TR_BitVector(TR::Region &region) : _numChunks(0), _chunks(NULL), _firstChunkWithNonZero(0), _lastChunkWithNonZero(-1), _growable(growable), _region(&region) { }
 
    // Construct a bit vector with a certain number of bits pre-allocated.
    // All bits are initially off.
    //
    TR_BitVector(int64_t initBits, TR_Memory * m, TR_AllocationKind allocKind = heapAlloc, TR_BitVectorGrowable growableOrNot = growable, TR_MemoryBase::ObjectType ot= TR_MemoryBase::BitVector)
       {
-      _allocationKind = allocKind;
       _chunks = NULL;
       _numChunks = getChunkIndex(initBits-1)+1;
       _firstChunkWithNonZero = _numChunks;
       _lastChunkWithNonZero = -1;
-      _trMemory = m;
+      _region = NULL;
+      switch (allocKind)
+         {
+         case heapAlloc:
+            _region = &(m->heapMemoryRegion());
+            break;
+         case stackAlloc:
+            _region = &(m->currentStackRegion());
+            break;
+         case persistentAlloc:
+            _region = NULL;
+            break;
+         default:
+            TR_ASSERT(false, "Unhandled allocation type!");
+         }
+      #ifdef TRACK_TRBITVECTOR_MEMORY
+      _memoryUsed=sizeof(TR_BitVector);
+      #endif
+      if (_numChunks)
+         {
+         if (_region)
+            {
+            _chunks = (chunk_t*)_region->allocate(_numChunks*sizeof(chunk_t));
+            }
+         else
+            {
+            TR_ASSERT(allocKind == persistentAlloc, "Should not allocate through trMemory except for persistent memory");
+            _chunks = (chunk_t*)m->allocateMemory(_numChunks*sizeof(chunk_t), persistentAlloc, ot);
```

Comment:
Nit: Could this please use jitPersistentAlloc for consistency? This appears to be the only place left in the using this form.